### DISCUSSION TITLE
The way I create the variables is somehow affecting the solver

### DISCUSSION THREAD

**Discussion by sarthakgaur** [google/or-tools#2792] [CP-SAT questions] [answered]
Language: Python
Solver: CP-SAT

We have a problem similar to the [employee scheduling](https://developers.google.com/optimization/scheduling/employee_scheduling) problem. The amount of variables that we need to create are quite large, and we have a lot of constraints.

I thought to optimize the way variables are being created. By optimizing, I mean to reduce the number of loops and function calls required to create the variables. I am not reducing the amount of variables I am creating, I am just changing the way I create them.

Initially, the solver was taking around 10 seconds to find a solution that was able to meet the objective. Yes, we do have an objective. So, our problem is more similar to [scheduling with shift requests](https://developers.google.com/optimization/scheduling/employee_scheduling#requests) example. I ran the solver multiple times and it was consistent around the 10 second mark.

After the optimization, the solver was able to find a solution that was able to meet the objective in less than 2 seconds. This was strange because I didn't make any changes to the model. We do have some tests that show that we are creating the same amount of variables after the optimization. It's not only the amount of variables, but we are creating exactly the same variables.

Then I did some more optimizations. All these optimizations are reducing the time it takes to create the variables as it is expected. But, this time the solver couldn't find a solution that was able to meet the objective even after several minutes. The tests again showed that we are creating exactly the same variables as before. This was strange, too.

Can someone explain why this is happening? Any help will be appreciated! Thanks!

**Comment by lperron** [comment, marked as answer]
It is well known that all solvers are sensitive to the order of the model variables and constraints. Especially, as the order of variables is a tie-break in the search.
